### ISSUE TITLE
Configurable table names (SQL)

### ISSUE THREAD

**Issue by strerene** [javers/javers#941]
Currently there is no way to change the default table names. 

- jv_global_id
- jv_commit
- jv_commit_property
- jv_snapshot

Providing a way of customizing these would be very nice.

**Comment by shirishpandharikar** [comment]
@bartoszwalacik I'm planning to work on this feature. Was thinking of providing a `TableConfiguration` class with a `TableConfigurationBuilder` where we can set the table names. We can then set this to the `SqlRepositoryConfiguration` via the `SqlReposirotyBuilder`. Another approach is to directly add these table names to the `SqlRepositoryConfiguration` but the constructor would be bloated. 

```java
TableConfiguration tableConf = TableConfigurationBuilder.tableConfiguration()
.withGlobalIdTableName("jav_global_id")
.withCommitTableName("jav_commit")
.withSnapshotTableName("jav_snapshot")
.withCommitPropertyTableName("jav_commit_property")
.build();

SqlRepositoryBuilder.sqlRepository().
withConnectionProvider(new ConnectionProvider() {
         public Connection getConnection() throws SQLException {
             return DriverManager.getConnection("jdbc:h2:mem:test;");
          }
}).
withDialect(DialectName.H2).
withTableConfiguration(tableConf).
build();
```
There will be a change in the `TableNameProvider` class to support this. Could you please let me know your thoughts on both the approaches? If this looks the right direction I will work on this and submit a PR. Will also look into the Mongodb module to support similar requirement.

**Comment by bartoszwalacik** [comment]
I think that nesting builders would be too complex, so:

```
SqlRepositoryBuilder.sqlRepository().
withConnectionProvider(new ConnectionProvider() {
         public Connection getConnection() throws SQLException {
             return DriverManager.getConnection("jdbc:h2:mem:test;");
          }
}).
withDialect(DialectName.H2).
withGlobalIdTableName("jav_global_id")
.withCommitTableName("jav_commit")
.withSnapshotTableName("jav_snapshot")
.withCommitPropertyTableName("jav_commit_property")
build();
```

**Comment by shirishpandharikar** [comment]
Sounds good. I will start making the changes. One thing I observed in the `FixedSchemaFactory` class is that the foreign key names and the constraint names are generated using the table names while the primary key names just append `jv_`. Also, the sequence names append `jv_`.

**Comment by bartoszwalacik** [comment]
Well, i guess the key names and sequence names should also be changed to reflect given table names.

**Comment by shirishpandharikar** [comment]
@bartoszwalacik I've made the required changes and submitted a PR. Please have a look at let me know your thoughts and comments.

**Comment by bartoszwalacik** [comment]
yes, i know, i will merge till frifay

**Comment by bartoszwalacik** [comment]
new props added in SqlRepository and in the starter:

```
  sqlGlobalIdTableName: 'cust_jv_global_id'
  sqlCommitTableName: 'cust_jv_commit'
  sqlSnapshotTableName: 'cust_jv_snapshot'
  sqlCommitPropertyTableName: 'cust_jv_commit_property'
```

**Comment by bartoszwalacik** [comment]
released in 5.10.5
